dft/constructor_injection.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python
+# -*- coding : utf-8 -*-
+import datetime
+
+"""
+Port of the Java example of "Constructor Injection" in
+"xUnit Test Patterns - Refactoring Test Code" by Gerard Meszaros
+(ISBN-10: 0131495054, ISBN-13: 978-0131495050)
+
+production code which is untestable:
+
+class TimeDisplay(object):
+
+    def __init__(self):
+        self.time_provider = datetime.datetime
+
+    def get_current_time_as_html_fragment(self):
+        current_time = self.time_provider.now()
+        current_time_as_html_fragment = "<span class=\"tinyBoldText\">" + current_time.hour + ":" + current_time.minute + "</span>"
+        return current_time_as_html_fragment
+"""
+
+class TimeDisplay(object):
+
+    def __init__(self, time_provider):
+        self.time_provider = time_provider
+
+    def get_current_time_as_html_fragment(self):
+        current_time = self.time_provider.now()
+        current_time_as_html_fragment = "<span class=\"tinyBoldText\">" + current_time + "</span>"
```

Comment:
Thanks but that's not what I meant. The idea is to avoid string concatenations completely since a) string are immutable and there is some overhead included, b) it is more idiomatic in Python to use `.format`. The idea looks like that: `"<span class=\"tinyBoldText\">{}{}</span>".format(current_time.hour, current_time.minute)`